cmd/root.go

```diff
@@ -111,6 +114,34 @@ func PreRun(cmd *cobra.Command, args []string) {
 func Run(c *cobra.Command, names []string) {
 	filter := filters.BuildFilter(names, enableLabel)
 	runOnce, _ := c.PersistentFlags().GetBool("run-once")
+	httpAPI, _ := c.PersistentFlags().GetBool("http-api")
+	
+	if httpAPI {
+		log.Println("Watchtower HTTP API started.")
+		
+		apiToken, _ := c.PersistentFlags().GetString("http-api-token")
+
+		http.HandleFunc("/v1/update", func(w http.ResponseWriter, r *http.Request){
+			log.Info("Updates triggered by HTTP API request.")
+			
+			_, err := io.Copy(os.Stdout, r.Body)
+			if err != nil {
+				log.Println(err)
+				return
+			}
+
+			if r.Header.Get("Token") != apiToken {
+				log.Println("Invalid token. Not updating.")
+				return
+			}
+
+			log.Println("Valid token found. Triggering updates.")
+			runUpdatesWithNotifications(filter)
```

Comment:
This might be a bit dangerous as we'd risk starting an update while another one is running. Maybe we need to add some kind of locking mechanism to this call to make it return something like "Already running an update" in case you try to trigger multiple updates?

Reply:
I've not paid attention to the lock in `runUpgradesOnSchedule`. I'll try to reproduce it in the API package.